framework/src/main/java/org/checkerframework/common/value/ValueTransfer.java

```diff
@@ -1563,6 +1563,7 @@ enum ConditionalOperators {
         AND;
     }
 
+    @SuppressWarnings("BooleanLiteral") // TODO: is there a performance difference?
```

Comment:
@Ao-senXiong In a small toy program, can you look whether the generated bytecode is the same if one uses `true` vs. `Boolean.TRUE`?

Reply:
This isn't performance critical code, so we should just do what is more readable. I'm just a bit surprised by the blanked recommendation to use the primitive type.

Reply:
The interesting question is what happens if there is a conversion. So can you try
````
boolean a = Boolean.TRUE; 
Boolean b = true;
````

That is closer to the question here, where the array has type `Boolean` and the fix suggests using `true`.

Reply:
You can look at the generated bytecode using `javap -v -cp checker/dist/checker.jar  org/checkerframework/common/value/ValueTransfer` and then looking for the static initializer.
The bytecode is different, so for now I will just suppress this warning.